File: RNApp/RNApp/ViewController.m
//
//  ViewController.m
//  RNApp
//
//  Created by 喻永权 on 2019/2/27.
//  Copyright © 2019年 喻永权. All rights reserved.
//

#import "ViewController.h"

#import <React/RCTRootView.h>
#import <React/RCTBundleURLProvider.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    UIButton *button = [UIButton new];
    button.frame = CGRectMake(0, 0, 80, 40);
    button.center = self.view.center;
    button.backgroundColor = [UIColor redColor];
    [button setTitle:@"RNDemo" forState:UIControlStateNormal];
    [self.view addSubview:button];
    [button addTarget:self action:@selector(press) forControlEvents:UIControlEventTouchUpInside];
}

- (void)press{
    NSLog(@"RN start...........");
    //第一种获取url的方式
    NSURL *url = [NSURL URLWithString:@"http://localhost:8081/ios.index.bundle?platform=ios"];
    //http://localhost:8081/ios.index.bundle?platform=ios
    NSLog(@"url:%@",url);
    
    
    //第二种获取url的方式
    NSURL *defaultUrl = [[RCTBundleURLProvider sharedSettings]jsBundleURLForBundleRoot:@"ios.index" fallbackResource:@"js"];
    //http://localhost:8081/ios.index.bundle?platform=ios&dev=true&minify=false
    NSLog(@"===>>%@",defaultUrl);
    
    RCTRootView *rootView = [[RCTRootView alloc]initWithBundleURL:defaultUrl moduleName:@"RNApp" initialProperties:nil launchOptions:nil];
    UIViewController *vc = [[UIViewController alloc]init];
    vc.view = rootView;
    [self presentViewController:vc animated:YES completion:nil];
}




- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Demo4-Camera/Demo4-Camera/ViewController.swift
//
//  ViewController.swift
//  Demo4-Camera
//
//  Created by 喻永权 on 2019/1/4.
//  Copyright © 2019年 喻永权. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var scrollView = UIScrollView()
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        UIApplication.shared.isStatusBarHidden = true
        scrollView.frame = CGRect.init(x: 0, y: 0, width: view.frame.width, height: view.frame.height)
        scrollView.contentSize = CGSize.init(width: view.frame.width * 3, height: view.frame.height)
        scrollView.bounces = false
        scrollView.isPagingEnabled = true
        scrollView.showsHorizontalScrollIndicator = false
        view.addSubview(scrollView)
        let leftVC = LeftController()
        let centerVC = CenterController()
        let rightVC = RightController()
        
        self.addChildViewController(leftVC)
        self.addChildViewController(centerVC)
        self.addChildViewController(rightVC)
        
        scrollView.addSubview(leftVC.view)
        scrollView.addSubview(centerVC.view)
        scrollView.addSubview(rightVC.view)
        
        //中间vc坐标
        var centerViewFrame : CGRect = centerVC.view.frame
        centerViewFrame.origin.x = view.frame.width
        centerVC.view.frame = centerViewFrame
        
        //右边vc坐标
        var rightViewFrame : CGRect = rightVC.view.frame
        rightViewFrame.origin.x = view.frame.width * 2
        rightVC.view.frame = rightViewFrame
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

